feat(vpc): doc to explain migration (#1834)
scaleway/docs-content: network/vpc/reference-content/vpc-migration.mdx

| VPC              | ---                                | VPC is introduced. One default VPC in each region is created automatically for all users. <br /><br /> Beta testers can create  regional Private Networks in their VPCs | All users have one default VPC per region automatically created for them. <br /><br /> Users can opt to create additional VPCs. <br /><br />All Private Networks now exist within a VPC.  |
| Public Gateways  | When attached to a Private Network, Public Gateways provide DHCP, NAT & SSH bastion for resources in the network.   |  No change | DHCP functionality leaves Public Gateways and is now built directly into Private Networks. <br /><br/> Public Gateways continue to offer NAT and SSH bastion. |

## Key concepts

> If you're unsure about some of the terminology in this document, refer to the sources below for more explanation beore continuing.

Reviewer: ```suggestion
If you are unsure about some of the terminology in this document, refer to the sources below for more explanation beore continuing.
```

> If you are unsure about some of the terminology in this document, refer to the sources below for more explanation beore continuing.

Reviewer 2: ```suggestion
If you are unsure about some of the terminology in this document, refer to the sources below for more explanation before continuing.
```

- The existing **Private Networks** section, available from the main menu (side bar) of the [console](https://console.scaleway.com/), will soon cease to be accessible. Instead, you should select **VPC** in the side bar, and then click on the relevant VPC, to access and manage your Private Networks.
- The DHCP dashboard for **Public Gateways** will no longer be available, as this functionality moves to Private Networks.

**API**
- The VPC v1 zoned API (which actually only concerns Private Networks and not VPC as it is now understood) will keep working, but will be deprecated and moved to read only. The VPC v2 API, covering both VPCs and Private Networks with regional scope and DHCP, will be the new default API. Note that on the [VPC API documentation](https://www.scaleway.com/en/developers/api/vpc/v2/), you can select v1 or v2 from the drop down menu at the top to navigate between the two versions.
- [DHCP reservations via the Public Gateway API](https://www.scaleway.com/en/developers/api/public-gateway/) will be deprecated but still available, to avoid breaking changes. In the future (no ETA), reservations should be available on Private Networks through IPAM.

**CLI**

> - As the CLI does not support multiple versions of an API, new and future versions will drop the VPC v1 API in favour of v2.

Reviewer 2: ```suggestion
- As the CLI does not support multiple versions of an API, new and future versions will drop the VPC v1 API in favor of v2.
```

- DHCP reservations via the Public Gateway will be deprecated but still available, to avoid breaking changes. In the future (no ETA), reservations should be available on Private Networks through IPAM.

**Terraform**
- Currently, a property exists on Private Network resources, to specify whether they are regional or not and allow use of the correct underlying API (VPC v1 or v2). This property will be deprecated once VPC is in GA.
- DHCP reservations via the Public Gateway will be deprecated but still available, to avoid breaking changes. In the future (no ETA), reservations should be available on Private Networks through IPAM.

## Why are we making these changes?

> Security and networking isn't easy. Mistakes can be made and details can be forgotten which can lead to conflicts in your configuration or serious security holes in your infrastructure. 

Reviewer: ```suggestion
Security and networking are not easy. Mistakes can be made and details can be forgotten which can lead to conflicts in your configuration or serious security holes in your infrastructure. 
```

We are aiming to make it as easy as possible for our customers to create a secure infrastructure, with resources reachable in a simple yet safe way, able to communicate with each other without having to be exposed on the public Internet.

> Each Scaleway Project will get a default VPC for each region. From there, it will only take a couple of clicks to create a Private Network with built-in DHCP in a VPC. No network configuration will be required, all the defaults are set up and ready to go. We even take care of making sure that each Private Network gets a large enough subnet that it won't overlap with that of any other Private Networks inside the same VPC.

Reviewer 2: ```suggestion
Each Scaleway Project will get a default VPC for each region. From there, it will only take a couple of clicks to create a Private Network with built-in DHCP in a VPC. No network configuration will be required, all the defaults are set up and ready to go. We even take care of making sure that each Private Network gets a large enough subnet that it will not overlap with that of any other Private Networks inside the same VPC.
```
